libass/ass.c

```diff
@@ -1357,28 +1333,28 @@ ASS_Track *ass_new_track(ASS_Library *library)
  */
 void ass_lazy_track_init(ASS_Library *lib, ASS_Track *track)
 {
-    if (track->PlayResX && track->PlayResY)
+    if (track->PlayResX > 0 && track->PlayResY > 0)
         return;
-    if (!track->PlayResX && !track->PlayResY) {
+    if (track->PlayResX <= 0 && track->PlayResY <= 0) {
         ass_msg(lib, MSGL_WARN,
                "Neither PlayResX nor PlayResY defined. Assuming 384x288");
         track->PlayResX = 384;
         track->PlayResY = 288;
     } else {
-        if (!track->PlayResY && track->PlayResX == 1280) {
+        if (track->PlayResY <= 0 && track->PlayResX == 1280) {
             track->PlayResY = 1024;
             ass_msg(lib, MSGL_WARN,
                    "PlayResY undefined, setting to %d", track->PlayResY);
-        } else if (!track->PlayResY) {
-            track->PlayResY = track->PlayResX * 3 / 4;
+        } else if (track->PlayResY <= 0) {
+            track->PlayResY = track->PlayResX * 3LL / 4;
             ass_msg(lib, MSGL_WARN,
                    "PlayResY undefined, setting to %d", track->PlayResY);
-        } else if (!track->PlayResX && track->PlayResY == 1024) {
+        } else if (track->PlayResX <= 0 && track->PlayResY == 1024) {
             track->PlayResX = 1280;
             ass_msg(lib, MSGL_WARN,
                    "PlayResX undefined, setting to %d", track->PlayResX);
-        } else if (!track->PlayResX) {
-            track->PlayResX = track->PlayResY * 4 / 3;
+        } else if (track->PlayResX <= 0) {
+            track->PlayResX = track->PlayResY * 4LL / 3;
```

Comment:
Note that this does not fix the int overflow on assignment if the value exceeds INT_MAX. As I noted in the pull request comment, we need to decide what to do in that case.